include/deal.II/lac/affine_constraints.h

```diff
@@ -2296,17 +2297,68 @@ inline AffineConstraints<number>::AffineConstraints(
 template <typename number>
 inline void
 AffineConstraints<number>::add_constraint(
-  const size_type                                  constrained_dof,
-  const std::vector<std::pair<size_type, number>> &dependencies,
-  const number                                     inhomogeneity)
+  const size_type                                      constrained_dof,
+  const ArrayView<const std::pair<size_type, number>> &dependencies,
+  const number                                         inhomogeneity)
 {
```

Comment:
Do you think it makes sense to move this function to the `affine_constraints.templates.h` function? I think it would fit better there; while there might be an overhead of 5-10 instructions by the function call that could possibly be elided, both `calculate_line_index()`, the `lines_cache.resize()` function and the final `for` loop are big enough that it should not matter, and compile time and library size would profit from not repeating the code.

Reply:
Sure, that seems sensible. So done!